packages/frontend/src/components/accounts/AccountFormAccountId.js

```diff
@@ -50,229 +51,208 @@ const InputWrapper = styled.div`
         }
     }
 `;
-class AccountFormAccountId extends Component {
-    state = {
-        accountId: this.props.defaultAccountId || '',
-        invalidAccountIdLength: false,
-        wrongChar: false
-    }
-
-    checkAccountAvailabilityTimer = null;
-    canvas = null;
-    suffix = createRef();
 
-    componentDidMount = () => {
-        const { defaultAccountId } = this.props;
-        const { accountId } = this.state;
-
-        if (defaultAccountId) {
-            this.handleChangeAccountId({ userValue: accountId });
+const AccountFormAccountId = (props) => {
+    const [accountId, setAccountId] = useState(props.defaultAccountId || '');
+    const [invalidAccountIdLength, setInvalidAccountIdLength] = useState(false);
+    const [wrongChar, setWrongChar] = useState(false);
+    const debouncedAccountId = useDebouncedValue(accountId, ACCOUNT_CHECK_TIMEOUT);
+    const canvas = useRef(null);
+    const suffix = createRef();
```

Comment:
We need to make sure we are using the correct standards, especially when converting now from Class to Functional components. A lot of things don't convert one-to-one. Please double-check these things, as there are often nuances which can make a big difference. Almost all "utility" things with Functional components will start with `use...`.

(https://blog.logrocket.com/complete-guide-react-refs/#useref-vs-create-ref)

Reply:
It helps to also look into the code sometimes. If you look into the code for `createRef()` you would see its not going to work out for a Functional Component paradigm.